src/main/java/com/pnu/dev/radioserviceapi/controller/admin/VideoAdminController.java

```diff
@@ -1,12 +1,73 @@
 package com.pnu.dev.radioserviceapi.controller.admin;
 
+import com.pnu.dev.radioserviceapi.exception.ServiceException;
+import com.pnu.dev.radioserviceapi.mongo.Video;
+import com.pnu.dev.radioserviceapi.service.VideoService;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Objects;
 
 @Controller
 @RequestMapping("/admin/videos")
 public class VideoAdminController {
 
-    // 1. add/remove recommended video
-    // 2. schedule video stream
+    private final VideoService videoService;
+
+    @Autowired
+    public VideoAdminController(VideoService videoService) {
+        this.videoService = videoService;
+    }
+
+    @GetMapping
+    public String findAll(Model model) {
+        List<Video> videos = videoService.findAll();
+        model.addAttribute("videos", videos);
+        return "videos/index";
+    }
+
+    @GetMapping("/add")
+    public String findVideoOnYoutube(@RequestParam(name = "link", required = false) String link, Model model) {
```

Comment:
Cannot you just use JS to substitute video url in embedded player, when the user enters new url?